sdcm/cluster.py

```diff
@@ -1078,6 +1079,8 @@ def uptime_changed():
         # wait until the reboot is executed
         wait.wait_for(func=uptime_changed, step=10, timeout=60*45, throw_exc=True)
 
+        collect_kallsyms_map(node=self)
```

Comment:
how long this function takes to run ? 

also why no run it once there a kernel stack found ? and upload it one per node / uptime  ?

I would rather not mess with the sequence of reboot, for uploading logs and such.

Reply:
It shouldn't collect the logs, only to "dump" the file or something like that.
I suggest changing the name of it from "collecting".

@fruch there will always be kernel stack.
The thing is that when one would want to decode it, it needs the stack from the relevant boot.
If the "dump" takes few seconds it should be fine.

Reply:
because these symbols are in memory this dump should take few milliseconds

Reply:
but it think it is not enough to only dump them, we must sync with the SCT Runner ASAP, so in case of sudden death of a node, we won't miss these symbols

Reply:
it taks half a second:
```
< t:2022-08-03 12:26:18,074 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "sudo cat /proc/kallsyms >& '/tmp/longevity-kallsyms-kallsyms-db-node-06347f39-2/kallsyms_20220803_122606'"...
< t:2022-08-03 12:26:18,617 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "sudo cat /proc/kallsyms >& '/tmp/longevity-kallsyms-kallsyms-db-node-06347f39-2/kallsyms_20220803_122606'" finished with status 0
```

Reply:
The method name doesn't have the word "log" in it but "map". I can rename it to "save_kallsyms_map"

Reply:
@yarongilor what's `log_entity.collect` do ? since you've showed the timing from the command running on the node only, and this `save_kallsyms_map` function does a bit more then that.

@fgelcer I think it would be safe enough to fetch it, on the first sight of kernel callstack, if you lost the node before reading the first callstack from the log, you have bigger problem then decoding it.

regardless, if there a flag for the enabling of the kernel callstack, it should also be used here

Reply:
@fruch :
1. The specific above call to `save_kallsysm_map` turned out to be redundant and was removed.
2. Roy's concern was about the dump itself not taking too long. (of course there shouldn't be too long wrapping overhead as well).
3. The full log of this method shows it takes 2 seconds to complete:
4. ``` < t:2022-08-03 16:15:21,526 f:logcollector.py l:819  c:sdcm.logcollector    p:INFO  > Saving kallsyms map from host: longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7
< t:2022-08-03 16:15:21,528 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "mkdir -p /tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7"...
< t:2022-08-03 16:15:21,978 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-08-03T16:15:21+00:00 longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7     !INFO | systemd-logind[1187]: New session 3 of user scylla-test.
< t:2022-08-03 16:15:21,979 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-08-03T16:15:21+00:00 longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7     !INFO | systemd[1]: Started Session 3 of user scylla-test.
< t:2022-08-03 16:15:22,463 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "mkdir -p /tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7" finished with status 0
< t:2022-08-03 16:15:22,463 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "uptime -s"...
< t:2022-08-03 16:15:22,470 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 2022-08-03 16:14:24
< t:2022-08-03 16:15:22,470 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "uptime -s" finished with status 0
< t:2022-08-03 16:15:22,471 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "sudo cat /proc/kallsyms >& '/tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424'"...
< t:2022-08-03 16:15:22,984 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-08-03T16:15:22+00:00 longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7   !NOTICE | sudo[2382]: scylla-test : TTY=unknown ; PWD=/home/scylla-test ; USER=root ; COMMAND=/usr/bin/cat /pro
c/kallsyms
< t:2022-08-03 16:15:22,984 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-08-03T16:15:22+00:00 longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7     !INFO | sudo[2382]: pam_unix(sudo:session): session opened for user root by (uid=0)
< t:2022-08-03 16:15:22,984 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-08-03T16:15:22+00:00 longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7     !INFO | sudo[2382]: pam_unix(sudo:session): session closed for user root
< t:2022-08-03 16:15:23,016 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "sudo cat /proc/kallsyms >& '/tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424'" finished with status 0
< t:2022-08-03 16:15:23,017 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "test -f '/tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424'"...
< t:2022-08-03 16:15:23,249 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > total,     345600085,   87752,   87752,   87752,    11.4,     9.0,    29.3,    40.5,    53.4,    86.2, 4070.0,  0.00140,      0,      0,       0,       0,       0,       0
< t:2022-08-03 16:15:23,519 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "test -f '/tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424'" finished with status 0
< t:2022-08-03 16:15:23,520 f:remote_base.py  l:195  c:RemoteLibSSH2CmdRunner p:DEBUG > Receive files (src) /tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424 -> (dst) /home/ubuntu/sct-results/20220803-140122-171879/longevity-10gb-3h-kallsyms-db-c
luster-5b3ba316/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms
< t:2022-08-03 16:15:23,520 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "rsync --version"...
< t:2022-08-03 16:15:23,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > rsync  version 3.1.3  protocol version 31
< t:2022-08-03 16:15:23,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Copyright (C) 1996-2018 by Andrew Tridgell, Wayne Davison, and others.
< t:2022-08-03 16:15:23,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Web site: http://rsync.samba.org/
< t:2022-08-03 16:15:23,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Capabilities:
< t:2022-08-03 16:15:23,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG >     64-bit files, 64-bit inums, 64-bit timestamps, 64-bit long ints,
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG >     socketpairs, hardlinks, symlinks, IPv6, batchfiles, inplace,
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG >     append, ACLs, xattrs, iconv, symtimes, prealloc
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > rsync comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > are welcome to redistribute it under certain conditions.  See the GNU
< t:2022-08-03 16:15:23,530 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > General Public Licence for details.
< t:2022-08-03 16:15:23,530 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "rsync --version" finished with status 0
< t:2022-08-03 16:15:23,532 f:local_cmd_runner.py l:58   c:LocalCmdRunner       p:DEBUG > Running command "rsync -L  --timeout=300 --rsh='/usr/bin/ssh -t -a -x  -o StrictHostKeyChecking=no -o UserKnownHostsFile=/tmp/tmp_4qfcsm6 -o BatchMode=yes -o ConnectTimeout=300 -o Serv
erAliveInterval=300 -l scylla-test -p 22 -i /home/ubuntu/.ssh/scylla-test' -az scylla-test@[10.142.0.62]:"/tmp/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms_20220803_161424" /home/ubuntu/sct-results/20220803-140122-171879/longevity-10gb-3h-kallsyms-db-cluster-5b3
ba316/longevity-10gb-3h-kallsyms-db-node-5b3ba316-0-7/kallsyms"...
< t:2022-08-03 16:15:23,534 f:config.py       l:273  c:fabric               p:DEBUG > File not found, skipping
```

Reply:
> @fgelcer I think it would be safe enough to fetch it, on the first sight of kernel callstack, if you lost the node before reading the first callstack from the log, you have bigger problem then decoding it.
> 
> regardless, if there a flag for the enabling of the kernel callstack, it should also be used here

the problem is that not always we have the callstack printed in the logs (only when we enable `kernel.perf_event_paranoid`), and we not always want it to be enabled anyway...
but this is true, that once we have such event, we could use it to fetch the `kallsyms`, but if we have already copy it in advance, would that be an issue?